README
jQuery-Mobile-DateBox
=====================

DateBox is a small, android-esk date picker, using as many of the native 
jQueryMobile controls as possible.

This is based in part on the work of Todd M. Horst, and his [Android Like Date Picker 2](http://toddmhorst.wordpress.com/2010/12/30/android-like-date-picker-with-jquery-mobile-2/)


DateBox Features
----------------

* Auto-parses text entered into form element on open [any valid date string in your browsers locale, uses javascript:Date()]
* Input items on DateBox are directly editable, allowing for quick date change.
* Auto-corrects out of bounds (calendar invalid) dates into the closest real date *(e.g. Feb 31st becomes March 2nd or 3rd depending on the year)*
* Attempts to position itself centered over the input, however, the top will never be less than 45px from the top boundry of the document, nor will the popup unessasarily extend the document.
* Click outside to close, configurable with option 'clickOutsideClose'
* Press ESC to close, configurable with option 'closeEscape' 
* .datebox('enable') and .datebox('disable') now fully functional.
* Auto-bind's to data-type='datebox' input elements
* Set data-options to override default widget options
* Outputs ISO 8601 Dates (YYYY-MM-DD)
* Localization possible with daysOfWeek and monthsOfYear options


Suggested Use
-------------

	// BEFORE loading jQM
	$( document ).bind( "mobileinit", function(){
		$.mobile.page.prototype.options.degradeInputs.date = 'text';
	});

### No Option Overrides:
		
	<input type="date" data-role="datebox" name="some_date" />
		
### With Option Overrides:
	
	<input type="date" data-role="datebox" name="some_date" data-options='{"buttonTheme": "b", "daysOfWeek" : ["Sun", "Mon", "Tue", "Wed", "Thr", "Fri", "Sat"]}' />


History.md

jQM-1.0.0 / 2011-12-01 
======================

  * Small Fixes: #124, #122, #121, #123
  * Merge pull request #118 from Skinney/master
  * Merge pull request #117 from frietsch/master
  * Add versioning to build process - speed up bugfix release process
  * Add option "forceInheritTheme" as a halfway meetign with #110
  * Missed a few use cases on the magic updating set button.  Referenced in #110
  * Fix data-theme inheritance for input element (something was wrong - it grabs the first parent it sees now, or uses "c" as default, same as jQM core
  * Add hidden combo of useInlineHideInput and centerWindow - seems to work as expected
  * Add i8n: it-IT italian
  * Add i8n: Norwegian Bokmaal translation.
  * No idea what the origin of that CSS line was, but with it, the dialog title bar is wider than the dialog body.
  
jQM-1.0rc3 / 2011-11-14
=======================

  * Merge pull request #116 from zmszaman/master
  * Merge pull request #115 from jrunge/master
  * Merge pull request #114 from agcolom/patch-1
  * Merge pull request #113 from jrunge/master
  * Merge pull request #111 from rodrigoalvesvieira/pt-BR
  * Merge pull request #108 from rodrigoalvesvieira/master
  * Fix calStartDay
  * Add Option highDatesAlt - a second highDates options (different theme)
  * passing date and datebox object to closeCallback()
  * changed _update() to hardreset() in _setOptions to properly update option changes
  * Massive overhaul of options matrix documentation - much more readable, and even mostly correct now
  * added jquery widget _setOption  function for changing datebox options after pagecreate
  * Update auto-attach method to make sure not to double-create if retriggered
  * Add a little css tweak for IE8 (maybe others) - Fixes #100
  * Add option "timeFormatOverride" to allow 12/24hr clock setting from data-options if needed
  * Add recuring events demo to site
  * Add create hook so that .trigger(create) will catch data-role="datebox" elements
  * Missed fieldsOrder on language change - add option "fieldsOrderOverride" to mimic old behavoir of this option
  * Update a few more things, make dateFormat part of i8n by default, unless set otherwise.  Closes #107, Closes #104
  * Add _getRecDays() function.  Probably going to expand this at some point, and maybe add it to the docs.
  * Forgot beforeToday option, fixes #106
  * Add i8n: Portuguese (Brazil).
  * Add i8n: Modern Greek (el) - Thanks to Giannis Kosmas
  
jQM-1.0rc2 / 2011-11-04 
=======================

  * BREAKING CHANGE! - Move i8n stuff to a saner place, allow multiple languages to be loaded at once.  BREAKING CHANGE!!
  * Add i8n: Japanese, work by moon dial of B$ - thanks
  * Add i8n: Swedish, Fix spelling errors in en/en-US.  Thanks to Henrik Ekselius for both
  * Add i8n: German, unknown author + google translate (needs work)
  * Add i8n: French
  * Add a few translations, and a CLI translation tool (just run it)
  * Move some stuff - intoduce the "hardreset" function (very cool)
  * Update build process to include copyright/license slug
  * Update license to CC 3.0 Attribution.
  * Merge pull request #98 from Skinney/master
  * default transition is now an option which can be set, instead of being hardcoded 'pop'
  * Add useInlineBlind option - Closes #65
  * Add afterToday and beforeToday on "extra" dates on calendars, Fixes #71
  
jQM-1.0rc1 / 2011-10-14 
=======================

  * Merge pull request #97 from dtao/patch-3
  * Merge pull request #82 from gga/master
  * Merge pull request #90 from evgenyneu/master
  * Merge pull request #83 from eromba/master
  * Merge pull request #89 from bjohn465/add-debug-flag
  * Merge pull request #79 from dtao/patch-2
  * Merge pull request #77 from gstroup/master
  * Merge pull request #66 from ShirtlessKirk/patch-1
  * Merge pull request #67 from dtao/patch-1
  * Merge pull request #75 from kpwebb/patch-1
  * Merge pull request #61 from forrest/master
  * Merge pull request #63 from aliok/add-default-value-option
  * Merge pull request #53 from aliok/disable-manual-input
  * Merge pull request #49 from aliok/master
  * Merge pull request #48 from evgenyneu/master
  * Add hook for orientation change, Closes #95
  * Fix for datebox in listview display.  Fixes #92
  * Add a few standard create hooks from jqm. Addresses #94
  * Add an enableDates option for calendar mode.  Closes #93
  * added check for close button to fix issue #84
  * Add beforeToday check on flipbox, fixes #85
  * Fix title issue on flipbox demo - fixes #80
  * Using media query instead of helper class min-width-480px
  * Added an optional open callback.
  * fixed issue #78: parent page removed from DOM when datebox widget is closed
  * fix to honor noAnimation option when opening calendar.
  * Add dateformat options "mmmm" and "dddd" - Closes #76
  * Work on #61 - Bring duration mode under new RegExp, elimate options: durationNoDays, durationNoSeconds, and durationShort. Add options: durationFormat. - THIS IS A BREAKING CHANGE!
  * Add rolloverMode option, to prevent the pickers from "rolling" the date/month/hour/minute/second. Closes  #57
  * Add some DOM cacheing control - so the parent page does not go away on us - fixes #58 and #59
  * Add modes: "flipbox" and "timeflipbox"
  * Rework slide mode - real draggable input now.  very, very spiffy
  * Make the new RegExp for reading date the default method - NOTE: this *may* be a breaking change for some!


index.php
<!DOCTYPE html> 
<html lang="en"> 
<head> 
	<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1" /> 
	<title>jQueryMobile - DateBox Demo</title>
	<link rel="stylesheet" href="http://code.jquery.com/mobile/1.0a4.1/jquery.mobile-1.0a4.1.min.css" />
	<link type="text/css" href="http://dev.jtsage.com/cdn/datebox/latest/jquery.mobile.datebox.min.css" rel="stylesheet" /> 
	
	<!-- NOTE: Script load order is significant! -->
	
	<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.5.1/jquery.min.js"></script> 
	<script type="text/javascript">
		$( document ).bind( "mobileinit", function(){
			$.mobile.page.prototype.options.degradeInputs.date = 'text';
		});	
	</script>
	<script type="text/javascript" src="http://code.jquery.com/mobile/1.0a4.1/jquery.mobile-1.0a4.1.min.js"></script>
	<script type="text/javascript" src="http://dev.jtsage.com/jquery.mousewheel.min.js"></script>
	<script type="text/javascript" src="http://dev.jtsage.com/cdn/datebox/latest/jquery.mobile.datebox.min.js"></script>
	<script type="text/javascript">
		 jQuery.extend(jQuery.mobile.datebox.prototype.options, {
             'experimentalReg': true
         });
	</script>
	
</head>
<body>
<div data-role="page" data-theme="a" id="main"> 
	<div data-role="header"> 
		<h1>jQueryMobile - DateBox Demo (1.0a4.1 base)</h1>
	</div>
	<div data-role="content" data-theme="c">
		<div data-role="collapsible">
			<h3>All Default Demo (src line: 28)</h3>
			<div data-role="fieldcontain">
				<label for="plugindefault">Plugin Defaults</label>
				<input value="2011-01-01" name="plugindefault" type="date" data-role="datebox" id="plugindefault" />
			</div>
		</div>
		<div data-role="collapsible" data-collapsed="true">
			<h3>Available Modes (src line: 35)</h3>
			<div data-role="fieldcontain"><!-- Date Picker, Android Styling -->
				<label for="defandroid">Date Picker, Android Style</label><input value="2011-01-01" name="defandroid" type="date" data-role="datebox" id="defandroid" />
			</div>
			<div data-role="fieldcontain"><!-- Date Picker, Calendar Display -->
				<label for="defcal">Date Picker, Calendar Style</label><input name="defcal" type="date" data-role="datebox" id="defcal" data-options='{"mode": "calbox"}'/>
			</div>
			<div data-role="fieldcontain"><!-- Date Picker, Slide Display -->
				<label for="slide">Date Picker, Slide Style</label><input value="2012-04-25" name="slide" type="date" data-options='{"mode": "slidebox"}' data-role="datebox" id="slide" />
			</div>
			<div data-role="fieldcontain"><!-- Time Picker, 12 Hour Clock -->
				<label for="time12">Time Picker, 12 Hour Clock</label><input value="8:03 AM" name="time12" type="text" data-options='{"mode": "timebox", "timeFormat": 12}' data-role="datebox" id="time12" />
			</div>
			<div data-role="fieldcontain"><!-- Time Picker, 24 Hour Clock -->
				<label for="time24">Time Picker, 24 Hour Clock</label><input value="10:22" name="time24" type="text" data-options='{"mode": "timebox"}' data-role="datebox" id="time24" />
			</div>
			<div data-role="fieldcontain"><!-- Duration Picker -->
				<label for="dur">Duration Picker</label><input value="5 Days, 04:03:02" name="dur" type="text" data-role="datebox" data-options='{"mode": "durationbox"}' id="dur" />
			</div>
		</div>
		
		<div data-role="collapsible" data-collapsed="true">
			<h3>Advanced Option Demos</h3><div data-role="collapsible-set">
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Forced Dialog Mode (src line: 61)</h3>
				<p>Forces the use of the jQM Dialog System, works in all modes</p>
				<div data-role="fieldcontain">
					<label for="adv1">Some Date</label>
					<input value="2011-01-01" name="adv1" type="date" data-role="datebox" id="adv1" data-options='{"useDialogForceTrue": true}'/>
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Forced Popup, Day-Month-Year display, Constrained Year (src line: 70)</h3>
				<p>Foces the use of the popup overlay mode (all modes), Input as Day-Month-Year (android only), with a minimum year of 2009 (android only)</p>
				<div data-role="fieldcontain">
					<label for="adv2">Some Date, After 2009 only.</label>
					<input value="2011-01-01" name="adv2" type="date" data-role="datebox" id="adv2" data-options='{"useDialogForceFalse": true, "minYear": 2009, "fieldsOrder": ["d", "m", "y"]}'/>
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Forced Popup No Animation mode (src line: 79)</h3>
				<p>Foces the use of the popup overlay mode (all modes) with no animation (all modes, limited to popup)</p>
				<div data-role="fieldcontain">
					<label for="adv3">Some Date</label>
					<input value="2011-01-01" name="adv3" type="date" data-role="datebox" id="adv3" data-options='{"mode": "calbox", "useDialogForceFalse": true, "noAnimation": true}'/>
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Modal Popup, Hidden Default, 1 Month Only (src line: 88)</h3>
				<p>Uses the modal popup for forced input (all modes), a hidden default date (all date modes), and shows only the current months dates (calendar only)</p>
				<div data-role="fieldcontain">
					<label for="adv4">Some Date</label><input name="adv4" type="date" data-role="datebox" id="adv4" data-options='{"mode": "calbox", "calShowOnlyMonth": true, "useModal": true, "defaultDate": "2010-01-01"}'/>
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Credit Card Expiration Entry (src line: 96)</h3>
				<p>Uses return formatting and field order limiting to input a month / year for a credit card (android only).</p>
				<div data-role="fieldcontain">
					<label for="adv5">Credit Card Expiration</label>
					<input value="1/2014" name="adv5" type="text" data-role="datebox" id="adv5" data-options='{"dateFormat": "mm/YYYY", "fieldsOrder": ["m", "y"]}' />
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>No button focus mode (src line: 105)</h3>
				<p>Demonstrates opening the date picker on focus rather than icon click or tap (all modes)</p>
				<div data-role="fieldcontain">
					<label for="adv6">Some Date</label>
					<input value="2011-01-01" name="adv6" type="date" data-role="datebox" id="adv6" data-options='{"mode": "calbox", "noButtonFocusMode": "true"}'/>
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Link Open (src line: 114)</h3>
				<p>Demonstrates opening the date picker with a <a href="#" id="plainlink">link</a></p>
				<script type="text/javascript">
					$('#plainlink').live('click', function() {
						$('#plainlinkdate').datebox('open');
					});
				</script>
				<div data-role="fieldcontain">
					<label for="plainlinkdate">Some Date</label>
					<input value="2011-01-01" name="plainlinkdate" type="date" data-role="datebox" id="plainlinkdate" />
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Date Range (src line: 128)</h3>
				<p>Demonstrates using date picker and a text box to enter a date range (customizable)</p>
				<script type="text/javascript">
					function updateEndDate() { // UPDATE END DATE FIELD
						// TO ACCESS THE DATE, GRAB FROM HERE
						var newdate = new Date($('#startdate').data('datebox').theDate);
						if ( newdate.getDate() ) { // VALID DATE IN FIRST INPUT
						
							// ADD THE VALUE OF #days SELECT TO DATE
							newdate.setDate(newdate.getDate() + parseInt($('#days').val()));
							
							// ADD PADDING ZEROS
							var padMonth = (( newdate.getMonth() < 9 ) ? "0" : "") + ( newdate.getMonth() + 1 ),
								padDay = (( newdate.getDate() < 10 ) ? "0" : "") + newdate.getDate();
							
							// DUMP OUTPUT INTO #enddate TEXT INPUT
							$('#enddate').val(newdate.getFullYear() + "-" + padMonth + "-" + padDay);
						}
					}
					
					$('#startdate').live('change', function() {
						// UPDATE WHEN THE START DATE IS CHANGED
						updateEndDate();
					});
					
					$('#days').live('change', function() {
						// UPDATE WHEN THE DATE RANGE IS CHANGED
						updateEndDate();
					});
				</script>
		
				<div data-role="fieldcontain">
					<label for="startdate">Start Date:</label>
					<input value="2011-01-01" name="startdate" type="date" data-role="datebox" id="startdate" data-options='{"useDialogForceFalse": true'/>
				</div>
				<div data-role='fieldcontain'>
					<label for='days'>Number of Days:</label>
					<select data-native-menu="false" name='days' id='days' >
						<option value='7' selected="selected">1 Week</option>
						<option value='6'>6 Days</option>
						<option value='5'>5 Days</option>
						<option value='4'>4 Days</option>
						<option value='3'>3 Days</option>
						<option value='2'>2 Days</option>
						<option value='1'>1 Day</option>
					</select>
				</div>
				<div data-role="fieldcontain">
					<label for="enddate">End Date:</label>
					<input id="enddate" type="text" />
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Hidden Input, Link Open (src line: 182)</h3>
				<p>Demonstrates opening a hidden date picker (value: <span id="hiddendateval"></span>) with a <a href="#" id="hlink">link</a>. Only works with forced dialog mode.</p>
				<script type="text/javascript">
					// NOTE: The event seemed to be attaching multiple times - threw a die in there to prevent 
					$('#hlink').die('click');
					$('#hlink').live('click', function(e) {
						$('#hiddenlinkdate').datebox('open');
					});
					$('#hiddenlinkdate').live('change', function() {
						$('#hiddendateval').text($('#hiddenlinkdate').val());
					});
				</script>
				<div data-role="fieldcontain" style="display: none !important;">
					<input value="2011-01-01" name="hiddenlinkdate" type="date" data-role="datebox" id="hiddenlinkdate" data-options='{"mode": "calbox", "useDialogForceTrue": true, "dateFormat": "MM/DD/YYYY"}' />
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Limit to today + 60 days (src line: 200)</h3>
				<p>Limit the allowed dates to at minimum today, and at maximum 60 days in the future. (both date modes)</p>
				<p>Note: the calendar mode limits users to viewing months with valid dates.</p>
				<div data-role="fieldcontain">
					<label for="limit">Android Style</label>
					<input name="limit" type="date" data-role="datebox" id="limit" data-options='{"afterToday": true, "maxDays": 60}'/>
				</div>
				<div data-role="fieldcontain">
					<label for="limit2">Calendar Style</label>
					<input name="limit2" type="date" data-role="datebox" data-options='{"minDays": 2, "maxDays": 60, "mode": "calbox"}' id="limit2" />
				</div>
			</div>

			<div data-role="collapsible" data-collapsed="true">
				<h3>Inline, hidden input display (src line: 214)</h3>
				<p>Display the date picker (any mode) inline.  Note that inline display will always take precendence.</p>
				<p>The current date is <span id="inlinedateval"></span></p>
				<script type="text/javascript">
					$('#inline').live('change', function() {
						$('#inlinedateval').text($('#inline').val());
					});
				</script>
				<div data-role="fieldcontain"><div id="isodiv"><style type="text/css">#isodiv>.ui-datebox-container { margin-right: auto; margin-left: auto; }</style>
					<label for="inline">Some Date</label>
					<input name="inline" type="date" data-role="datebox" data-options='{"useInline": true, "useInlineHideInput": true, "mode": "calbox"}' id="inline" />
				</div></div>
			</div>

			<div data-role="collapsible" data-collapsed="true">
				<h3>Blacklisted Days and Dates (src line: 229)</h3>
				<p>Blacklist days (day of week), or dates (yyyy/mm/dd) - Calendar mode only</p>
				<p>Date Format:  must be a simple array of ISO style dates, i.e.: ["2011-01-01", "2011-01-02"] etc.</p>
				<p>Example shows No Tuesdays and a few days in May and June 2011 disabled.</p>

				<div data-role="fieldcontain">
					<label for="blacklist2">Some ISO Date</label>
					<input value="2011-05-01" name="blacklist2" type="date" data-role="datebox" id="blacklist2" data-options='{"mode": "calbox", "blackDays": [2], "blackDates": ["2011-05-31", "2011-06-01", "2011-05-03","2011-05-04","2011-05-05"]}'/>
				</div>
			</div>

			<div data-role="collapsible" data-collapsed="true">
				<h3>Limit to today + 10 days, then based on start. (src line: 245)</h3>
				<p>The first date is limited from "today" to 10 days in the future.</p>
				<p>The second date is limited from the first date, to 10 days after the first date</p>
				<p>Note: yes, apparently some of the options can be changed dynamicly.  cool</p>
				<p>Note #2: minDays will take a negative number.   *very* cool.</p>
				<script type="text/javascript">
					$('#rlimit').live('change', function() {
						$('#rlimit2').val($('#rlimit').val());
						var temp = new Date(),
							diff = parseInt(($('#rlimit').data('datebox').theDate - temp) / ( 1000 * 60 * 60 * 24 ));
							diffstrt = (diff * -1)-1; // If you want a minimum of 1 day between, make this -2 instead of -1
							diffend = diff + 11; // Why 11 instead of 10?  No idea...
					
						$('#rlimit2').data('datebox').options.minDays = diffstrt;
						$('#rlimit2').data('datebox').options.maxDays = diffend;
					});
				</script>
				<div data-role="fieldcontain">
					<label for="rlimit">Beginning Date</label>
					<input name="rlimit" type="date" data-role="datebox" id="rlimit" data-options='{"mode": "calbox", "afterToday": true, "maxDays": 10}'/>
				</div>
				<div data-role="fieldcontain">
					<label for="rlimit2">Ending Date</label>
					<input name="rlimit2" type="date" data-role="datebox" data-options='{"minDays": -10, "maxDays": 10, "mode": "calbox"}' id="rlimit2" />
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Week Mode (src line: 273)</h3>
				<p>Use the first day of week (configurable day, set to monday [1] in demo) mode.  Calendar only.</p>
				
				<div data-role="fieldcontain">
					<label for="week">Some Monday</label>
					<input value="2011-01-01" name="week" type="date" data-role="datebox" id="week" data-options='{"mode": "calbox", "calWeekModeFirstDay": 1, "calWeekMode": true}' />
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Different Start Day (src line: 279)</h3>
				<p>Start the calendar on a day other than sunday (monday shown). Calendar only.</p>
				
				<div data-role="fieldcontain">
					<label for="shiftday">Some Date</label>
					<input name="shiftday" type="date" data-role="datebox" data-options='{"mode": "calbox", "calStartDay": 1}' id="shiftday" />
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Linked Inputs [callbacks] (src line: 289)</h3>
				<p>While it is almost always enough to listen for the 'change' event on the datebox input, occasionly you may need to trigger an event only when DateBox is done.  For this purpose, DateBox provides a very simple callback mechanism.  The following date and time inputs are linked so that a user might need to supply both.</p>
				
				<div data-role="fieldcontain">
					<label for="linkdate">Some Date</label>
					<input name="linkdate" type="date" data-role="datebox" data-options='{"closeCallback": "$(\"#linktime\").datebox(\"open\");"}' id="linkdate" />
					<label for="linktime">Some Time</label>
					<input name="linktime" type="text" data-role="datebox" data-options='{"mode": "timebox", "timeFormat": 12, "noButton": true}' id="linktime" />
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Duration Picker with options (src line: 301)</h3>
				<p>A duration picker with 15 minute stepping, and no DAY display</p>
				<div data-role="fieldcontain">
					<label for="dur2">Some Duration</label>
					<input name="dur2" type="text" data-role="datebox" data-options='{"mode": "durationbox", "durationNoDays":true, "durationSteppers":{"d":1,"h":1,"i":15,"s":1}}' id="dur2" />
				</div>
			</div>
			
			<div data-role="collapsible" data-collapsed="true">
				<h3>Highlighted Dates (src line: 315)</h3>
				<p>Highlight dates (yyyy/mm/dd) - Calendar mode only</p>
				<p>Date Format:  must be a simple array of ISO style dates, i.e.: ["2011-01-01", "2011-01-02"] etc.</p>
				<p>Example shows a few days in June 2011 highlighted, along with "today" highlighting toggeled off.</p>

				<div data-role="fieldcontain">
					<label for="high1">Some ISO Date</label>
					<input value="2011-06-05" name="high1" type="date" data-role="datebox" id="high1" data-options='{"mode": "calbox", "highDates": ["2011-06-01", "2011-06-20","2011-06-21"], "calHighToday": false}'/>
				</div>
			</div>
			
		</div></div>
		<div data-role="collapsible" data-collapsed="true">
			<h3>Other Demos</h3>
			<p>These other demos are my working pages - they break often.</p>
			<p><a data-role="button" rel="external" href="/jQM-DateBox/tests/dev.php">Very simple devel overview.</a></p>
			<p><a data-role="button" rel="external" href="/jQM-DateBox/tests/testing.php">This page, linked against jQM latest sources instead.</a></p>
		</div>
		<div data-role="collapsible">
			<!-- NOTE: When running locally, just nuke this section, or go grab PHP Markdown -->
			<h3>Readme File</h3>
			<?php
				include_once "../markdown.php";
				$md_text = file_get_contents("README.md");
				echo Markdown($md_text);
			?>
		</div>
	</div>
	<div data-role="footer">
		<a href="https://github.com/jtsage/jquery-mobile-datebox">GitHub Source</a><a rel='external' href="http://dev.jtsage.com/blog/">Blog</a><a href="mailto:jtsage+datebox@gmail.com">Contact</a>
	</div>
</div>
</html>
